tools/apigee-proxy-modifier-validator/modify_proxies.py

```diff
@@ -0,0 +1,980 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module providing a set of functions to modify
+and validate apigee proxies."""
+
+
+import argparse
+import difflib
+import json
+import logging
+import os
+import re
+import sys
+import tempfile
+import zipfile
+from pathlib import Path
+import multiprocessing
+
+import google.auth  # pylint: disable=import-error
+import google.auth.transport.requests  # pylint: disable=import-error
+import requests  # pylint: disable=import-error
+import tomlkit  # pylint: disable=import-error
+from lxml import etree  # nosec B410 pylint: disable=import-error
+
+# Setup basic logging configuration
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - [%(funcName)s] %(message)s',
```

Comment:
`format` will be set once in the config, instead of in each logging method, resulting in each log line having the same timestamp.